Firmware/Filament_sensor.cpp

```diff
@@ -278,7 +278,9 @@ void IR_sensor_analog::voltUpdate(uint16_t raw) { // to be called from the ADC I
 }
 
 uint16_t IR_sensor_analog::getVoltRaw() {
-    ATOMIC_BLOCK(ATOMIC_RESTORESTATE) { return voltRaw; }
+    uint16_t ret;
+    ATOMIC_BLOCK(ATOMIC_RESTORESTATE) { ret = voltRaw; }
```

Comment:
I need David's approval on these. I think this is how I did them initially.

Reply:
I checked the generated asm is identical. The alternative would be to add a dummy "return", which is a lot uglier IMHO.

Reply:
If the generated code is identical, then I have 0 problems with this change. Still, I'd let David approve this part since he changed it after me

Reply:
I agree